conf/charts/training/templates/job.yaml

```diff
@@ -5,18 +5,14 @@ metadata:
   labels:
     app: {{ .Values.app_name }}
 spec:
-  selector:
-      matchLabels:
-        app: {{ .Values.app_name }}
-  parallelism: 0
-  completions: null
+  parallelism: {{ .Values.parallelism }}
+  completions: 2147483647
```

Comment:
See https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/#parallel-jobs

I think you want to set completions to null for infinite

Reply:
And set parallelism to some positive non zero integer

Reply:
We initialize the job with a parallelism for 0 (to disable the job), then when our `autoscaler` sees a Redis training entry, it adjusts the parallelism to 1 (or to the number of training hashes) so that the jobs will deploy.  We use this pattern to minimize the time the GPUs are up.

For completions:
I had it initially set to null so that it will continue to run forever, but that is not the behaviour I found.  with `completions: null`, after the first job was run, I was not able to deploy more jobs.  I'm not sure why, that is not what I would expect to happen.  However, by setting the completions to a huge value, the jobs are being deployed as expected.  Seems like a hacky workaround, but for our use-case, I don't envision hitting the limit of 2 billion.